File: content/public/common/url_utils.cc
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2013 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "content/public/common/url_utils.h"

#include "build/build_config.h"
#include "content/common/url_schemes.h"
#include "content/public/common/url_constants.h"

namespace content {

bool HasWebUIScheme(const GURL& url) {
  return url.SchemeIs(kChromeDevToolsScheme) ||
         url.SchemeIs(kChromeUIScheme);
}

bool IsSavableURL(const GURL& url) {
  for (auto& scheme : GetSavableSchemes()) {
    if (url.SchemeIs(scheme))
      return true;
  }
  return false;
}

}  // namespace content

File: chrome/android/java/src/org/chromium/chrome/browser/payments/UrlUtils.java
FASTVC-BENCH-CORPUS:chromium-main-v1-943b94ae-1c74-4335-94fa-ceb4d277cea8
// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.payments;

import org.chromium.components.embedder_support.util.UrlConstants;
import org.chromium.url.GURL;

/** URL utilities. */
public class UrlUtils {
    /**
     * Returns false for invalid URL format or a relative URI.
     *
     * @param url The payment method name.
     * @return TRUE if given url is valid and not a relative URI.
     */
    public static boolean isURLValid(GURL url) {
        return url != null && url.isValid() && !url.getScheme().isEmpty()
                && (UrlConstants.HTTPS_SCHEME.equals(url.getScheme())
                        || UrlConstants.HTTP_SCHEME.equals(url.getScheme()));
    }

    private UrlUtils() {}
}
